### PR TITLE
customize: custom fields

### LINKED ISSUE #403: Custom fields docs
- Brief feature preview in v10 Release notes
- Full "tutorial" under Customize
  - We also need a "reference" for the available fields and UI widgets
- Under "Develop > Topic guides"?

### PR DESCRIPTION
**Pull request by slint**
Closes #403.

### CODE REVIEW

File: docs/customize/custom_fields/records.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+After configuring your custom fields, you have to update the corresponding search mappings. Otherwise those records will fail to be indexed, and facets/search will not work on them. This is easily done via the `invenio custom-fields ...` command in the following manner:
+
+```bash
+# for creating any missing custom field mappings for records
+invenio rdm-records custom-fields init [-f field_name] [-f field_name]
+```
+
+!!! info
+    Note that if the custom fields are configured **before** the InvenioRDM instance has been setup, this is the `invenio-cli services setup` command has never been run, the custom fields will be added by it and no command needs to be run.
+
```
Comment (ppanero) [inline docs/customize/custom_fields/records.md:241]: add to the note:

> If the custom fields contain required fields, demo records will fail. Therefore the setup has to be done as invenio-cli services setup --no-demo-records. They might still fail in the landing page UI, unless the default is changed.

Comment (zzacharo) [reply docs/customize/custom_fields/records.md:310]: I have documented it

File: docs/develop/topics/custom_fields.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+}} {% endfor %}
+```
+
+## Applying configuration
+
+We have defined our field, so now we need to add it to the `RDM_CUSTOM_FIELDS` and `RDM_CUSTOM_FIELDS_UI` configuration:
+
+```python
+RDM_CUSTOM_FIELDS = [
+    Experiments(name="experiments")
```
Comment (ppanero) [inline docs/develop/topics/custom_fields.md:163]:
```suggestion
    ExperimentsCF(name="experiments")
```

**Review by ppanero — COMMENTED** [review]
Few tiny changes, I'll apply and merge

File: docs/develop/topics/custom_fields.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+We will start by defining a new custom field type.
+
+```python
+from invenio_records_resources.services.custom_fields import BaseCF
+from marshmallow_utils.fields import SanitizedUnicode
+
+class Experiments(BaseCF):
+    """List of experiments with extra information."""
+
+    def __init__(self, name, **kwargs):
```
[the remark this answers is not in the document]
Comment (ppanero) [reply docs/develop/topics/custom_fields.md:37]: @zzacharo I've modified the example to use the `BaseListCF`, modify if Im not correct.

**Pull request merged**